File: app/src/data/payment-provider.ts
export enum PaymentProvider {
    ePayBL = 'epaybl', // Must be in sync with data in application.properties
    pmPayment = 'pmpayment', // Must be in sync with data in application.properties
    GiroPay = 'giropay', // Must be in sync with data in application.properties
}

export const PaymentProviderLabels: Record<PaymentProvider, string> = {
    [PaymentProvider.ePayBL]: 'ePayBL', // Must be in sync with data in application.properties
    [PaymentProvider.pmPayment]: 'pmPayment', // Must be in sync with data in application.properties
    [PaymentProvider.GiroPay]: 'giropay', // Must be in sync with data in application.properties
};

File: src/main/java/de/aivot/GoverBackend/enums/PaymentProvider.java
package de.aivot.GoverBackend.enums;

import com.fasterxml.jackson.annotation.JsonValue;
import de.aivot.GoverBackend.lib.models.Identifiable;

public enum PaymentProvider implements Identifiable<String> {
    ePayBL("epaybl", "ePayBL"),
    pmPayment("pmpayment", "pmPayment"),
    giroPay("giropay", "giropay"),
    ;

    private final String key;
    private final String label;

    PaymentProvider(String key, String label) {
        this.key = key;
        this.label = label;
    }

    @Override
    @JsonValue
    public String getKey() {
        return key;
    }

    public String getLabel() {
        return label;
    }

    @Override
    public boolean matches(Object other) {
        return key.equals(other);
    }
}
